src/core/border_router/routing_manager.cpp

```diff
@@ -1943,11 +1961,44 @@ void RoutingManager::OmrPrefixManager::Evaluate(void)
 
     DetermineFavoredPrefix();
 
-    // Decide if we need to add or remove our local OMR prefix.
+    // Determine the local prefix and remove outdated prefix published by us.
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+    if (Get<RoutingManager>().mPdPrefixManager.HasPreferredPrefix())
+    {
+        // Only preferred prefix will be published. Deprecated prefix will be ignored since they won't provide internet
+        // connectivity usually.
```

Comment:
Can remove this?
Since we no longer keep track of deprecated prefix, we can simplfiy and use `HasPrefix()`?